protocol/x/vault/keeper/withdraw_test.go

```diff
@@ -0,0 +1,430 @@
+			// posterior_leverage = 3 / (1 - 0.000001) ~= 3.000004
+			// integral
+			// = skew_antiderivative(skew_factor, posterior_leverage) - skew_antiderivative(skew_factor, leverage)
+			// = 2 * 3.000004^2 + 2^2 * 3.000004^3 / 3 - (2 * 3^2 + 2^2 * 3^3 / 3)
+			// ~= 54.000194 - 54
+			// = 0.000194
+			// average_skew = 0.000194 / (3.000004 - 3) = 48.5
+			// slippage = spread * (1 + average_skew) * leverage
+			// = 0.003 * (1 + 48.5) * 3
+			// = 0.4455
+			// slippage = min(0.4455, leverage * imf)
+			// = min(0.4455, 3 * 0.2) = 0.4455
+			expectedSlippagePpm: big.NewInt(445_500),
+		},
+		"Success: leverage 3, skew 2, spread 0.003, withdraw 10%": {
+			skewFactorPpm:        2_000_000,
+			spreadMinPpm:         3_000,
+			spreadBufferPpm:      1_500,
+			minPriceChangePpm:    1_000,
+			assetQuoteQuantums:   big.NewInt(-2_000_000_000), // -2,000 USDC
+			positionBaseQuantums: big.NewInt(1_000_000_000),  // 1 ETH
+			vaultId:              testVaultId,
+			withdrawalPortionPpm: big.NewInt(100_000), // 10%
+			// open_notional = 1_000_000_000 * 10^-9 * 3_000 * 10^6 = 3_000_000_000
+			// leverage = 3_000_000_000 / (-2_000_000_000 + 3_000_000_000) = 3
+			// posterior_leverage = 3 / (1 - 0.1) = 3.333333
+			// integral
+			// = skew_antiderivative(skew_factor, posterior_leverage) - skew_antiderivative(skew_factor, leverage)
+			// = 2 * 3.333333^2 + 2^2 * 3.333333^3 / 3 - (2 * 3^2 + 2^2 * 3^3 / 3)
+			// = 71.604919 - 54
+			// = 17.604919
+			// average_skew = 17.604919 / (3.333333 - 3) ~= 52.814810
+			// slippage = spread * (1 + average_skew) * leverage
+			// = 0.003 * (1 + 52.814810) * 3
+			// = 0.48433329
+			// round up to 0.484334
+			// slippage = min(0.484334, leverage * imf)
+			// = min(0.484334, 3 * 0.2) = 0.484334
+			expectedSlippagePpm: big.NewInt(484_334),
+		},
+		"Success: leverage 3, skew 2, spread 0.003, withdraw 50%": {
+			skewFactorPpm:        2_000_000,
+			spreadMinPpm:         2_000,
+			spreadBufferPpm:      1_500,
+			minPriceChangePpm:    1_500,
+			assetQuoteQuantums:   big.NewInt(-2_000_000_000), // -2,000 USDC
+			positionBaseQuantums: big.NewInt(1_000_000_000),  // 1 ETH
+			vaultId:              testVaultId,
+			withdrawalPortionPpm: big.NewInt(500_000), // 50%
+			// open_notional = 1_000_000_000 * 10^-9 * 3_000 * 10^6 = 3_000_000_000
+			// leverage = 3_000_000_000 / (-2_000_000_000 + 3_000_000_000) = 3
+			// posterior_leverage = 3 / (1 - 0.5) = 6
+			// integral
+			// = skew_antiderivative(skew_factor, posterior_leverage) - skew_antiderivative(skew_factor, leverage)
+			// = 2 * 6^2 + 2^2 * 6^3 / 3 - (2 * 3^2 + 2^2 * 3^3 / 3)
+			// = 360 - 54
+			// = 306
+			// average_skew = 306 / (6 - 3) = 102
+			// slippage = spread * (1 + average_skew) * leverage
+			// = 0.003 * (1 + 102) * 3
+			// = 0.927
+			// slippage = min(0.4455, leverage * imf)
+			// = min(0.927, 3 * 0.2) = 0.6
+			expectedSlippagePpm: big.NewInt(600_000),
+		},
+		"Success: leverage 3, skew 2, spread 0.003, withdraw 99.9999%": {
```

Comment:
nit: Test withdrawing 100% of vault?

Reply:
a test case above is for withdrawing 100%